### DISCUSSION TITLE
Which module and class emits the java hint ';' expected

### DISCUSSION THREAD

**Discussion by Hervian** [apache/netbeans#6961] [Q&A]
I am playing around with javac's [Compiler Plugin API](https://docs.oracle.com/en/java/javase/17/docs/api/jdk.compiler/com/sun/source/util/Plugin.html) but the Netbeans editor does not seem to pick it up. In my experiment I am trying to create a plugin that adds missing semicolons.

My plugin works fine when running "Clean and Build" on the project, and also when running "mvn clean compile" from a terminal. That is: the plugin is activated and called on the PARSE event.
But in the editor some module (java hints?) puts the line with the missing semicolon in error. 

So I am wondering if one can somehow get the editor to find the plugin (adding -Xplugin to my compiler args does not work, nor does using the plugin APIs autodiscovery). Also, I am wondering which class/module emits the "';' expected" error. It really sounds like an error from the java compiler but perhaps it comes from some static code analysis tool being run?

I should add that adding the plugin programmatically via an AbstractProcessor does not work either. The plugin is indeed being invoked (the tasklistener registered is), also for PARSE events after edits forces a new compilation, but it still does not remove the missing semicolon error in the editor.

<img width="158" alt="Screenshot 2024-01-16 154119" src="https://github.com/apache/netbeans/assets/7109864/37575db0-1f5d-4acc-9f26-d2fbaad82498">

**Comment by mbien** [comment]
I don't think this is a NB hint in the traditional sense. NB itself uses the javac API and this is likely simply the msg of the javac error converted into the editor annotation. A quick test showed that javac emits the very same `';' expected` compiler error.

**Reply by Hervian** [reply]
Thanks for your reply.

Having looked more at it I think you are right. I think it comes from nb-javac which seems to be used by the editor when compiling the source files. These classes seems to be slightly modified javac classes.
